src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -301,6 +300,22 @@ private void RegisterLocalServices()
 
                 return new ConfigurationProvider(legacyConfigPathProvider, connectedModeConfigPathProvider, bindingSerializer, legacyPostSaveOperation);
             }));
+
+            var projectNameTestProjectIndicator = new Lazy<ILocalService>(() => new ProjectNameTestProjectIndicator(Logger));
+            this.localServices.Add(typeof(ITestProjectRegexSetter), projectNameTestProjectIndicator);
+
+            this.localServices.Add(typeof(ITestProjectIndicator), new Lazy<ILocalService>(() =>
+            {
+                var buildPropertyTestProjectIndicator = new BuildPropertyTestProjectIndicator(this);
+                var testProjectIndicators = new List<ITestProjectIndicator>
+                {
+                    new ProjectKindTestProjectIndicator(this),
+                    projectNameTestProjectIndicator.Value as ITestProjectIndicator,
+                    new ProjectCapabilityTestProjectIndicator(this)
+                };
+
+                return new TestProjectIndicator(buildPropertyTestProjectIndicator, testProjectIndicators);
```

Comment:
Does this need to be a service? It looks like it is only used by the `ProjectSystemFilter`.

In general, let's try to avoid adding new services if they are not required - it just adds complexity and makes it harder to track dependencies.